tikv-client/src/main/java/com/pingcap/tikv/codec/MyDecimal.java

```diff
@@ -151,8 +150,8 @@ public int fromBin(int precision, int frac, int[] bin) {
     }
 
     this.negative = mask != 0;
-    this.digitsInt = wordsInt * digitsPerWord + leadingDigits;
-    this.digitsFrac = wordsFrac * digitsPerWord + trailingDigits;
+    this.digitsInt = (byte)(wordsInt * digitsPerWord + leadingDigits);
+    this.digitsFrac = (byte)(wordsFrac * digitsPerWord + trailingDigits);
```

Comment:
type conversion seems unnecessary. Digits of int and fraction can never go beyond 127. tidb does this simply because it is static type language. It has to do such conversion otherwise compiler will not be happy.

Reply:
To keep the same behavior as in TiDB which does a cap.